var/spack/repos/builtin/packages/paraview/package.py

```diff
@@ -65,61 +71,67 @@ class Paraview(Package):
-
-            feature_args = std_cmake_args[:]
-            feature_args.append(
-                '-DPARAVIEW_BUILD_QT_GUI:BOOL=%s' % feature_to_bool('+qt'))
-            feature_args.append('-DPARAVIEW_ENABLE_PYTHON:BOOL=%s' %
-                                feature_to_bool('+python'))
-            if '+python' in spec:
-                feature_args.append(
-                    '-DPYTHON_EXECUTABLE:FILEPATH=%s/bin/python'
-                    % spec['python'].prefix)
-            feature_args.append('-DPARAVIEW_USE_MPI:BOOL=%s' %
-                                feature_to_bool('+mpi'))
-            if '+mpi' in spec:
-                feature_args.append(
-                    '-DMPIEXEC:FILEPATH=%s/bin/mpiexec' % spec['mpi'].prefix)
-            feature_args.append(
-                '-DVTK_ENABLE_TCL_WRAPPING:BOOL=%s' % feature_to_bool('+tcl'))
-            feature_args.append('-DVTK_OPENGL_HAS_OSMESA:BOOL=%s' %
-                                feature_to_bool('+osmesa'))
-            feature_args.append('-DVTK_USE_X:BOOL=%s' %
-                                nfeature_to_bool('+osmesa'))
-            feature_args.append(
-                '-DVTK_RENDERING_BACKEND:STRING=%s' %
-                feature_to_bool('+opengl2', 'OpenGL2', 'OpenGL'))
-
-            feature_args.extend(std_cmake_args)
-
-            if 'darwin' in self.spec.architecture:
-                feature_args.append('-DVTK_USE_X:BOOL=OFF')
-                feature_args.append(
-                    '-DPARAVIEW_DO_UNIX_STYLE_INSTALLS:BOOL=ON')
-
-            cmake('..',
-                  '-DCMAKE_INSTALL_PREFIX:PATH=%s' % prefix,
-                  '-DBUILD_TESTING:BOOL=OFF',
-                  '-DVTK_USE_SYSTEM_FREETYPE:BOOL=ON',
-                  '-DVTK_USE_SYSTEM_HDF5:BOOL=OFF',
-                  '-DVTK_USE_SYSTEM_JPEG:BOOL=ON',
-                  '-DVTK_USE_SYSTEM_LIBXML2:BOOL=ON',
-                  '-DVTK_USE_SYSTEM_NETCDF:BOOL=OFF',
-                  '-DVTK_USE_SYSTEM_TIFF:BOOL=ON',
-                  '-DVTK_USE_SYSTEM_ZLIB:BOOL=ON',
-                  *feature_args)
-            make()
-            make('install')
+        if version < Version('5.1.0'):
+            return self._urlfmt.format(version.up_to(2), version, '-source')
+        else:
+            return self._urlfmt.format(version.up_to(2), version, '')
+
+    def cmake_args(self):
+        """Populate cmake arguments for ParaView."""
+        spec = self.spec
+
+        def variant_bool(feature, on='ON', off='OFF'):
+            """Ternary for spec variant to ON/OFF string"""
+            if feature in spec:
+                return on
+            return off
+
+        def nvariant_bool(feature):
+            """Negated ternary for spec variant to OFF/ON string"""
+            return variant_bool(feature, on='OFF', off='ON')
+
+        rendering = variant_bool('+opengl2', 'OpenGL2', 'OpenGL')
+        includes  = variant_bool('+plugins')
+
+        cmake_args = std_cmake_args[:]
```

Comment:
You don't need this line anymore. `std_cmake_args` are already added the the `cmake` call in `CMakePackage`. See `spack edit -b cmake` for details.